internal/services/amlfilesystem/aml_file_system_resource.go

```diff
@@ -0,0 +1,512 @@
+			}
+
+			identity, err := expandAMLFileSystemIdentity(model.Identity)
+			if err != nil {
+				return err
+			}
+
+			properties := &amlfilesystems.AmlFilesystem{
+				Location: location.Normalize(model.Location),
+				Identity: identity,
+				Properties: &amlfilesystems.AmlFilesystemProperties{
+					Hsm:                expandAMLFileSystemHsmSetting(model.HsmSetting),
+					EncryptionSettings: expandAMLFileSystemKeyEncryptionKey(model.KeyEncryptionKey),
+					MaintenanceWindow:  expandAMLFileSystemMaintenanceWindowForCreate(model.MaintenanceWindow),
+					FilesystemSubnet:   model.SubnetId,
+					StorageCapacityTiB: float64(model.StorageCapacityInTb),
+				},
+				Sku: &amlfilesystems.SkuName{
+					Name: pointer.To(model.SkuName),
+				},
+				Zones: pointer.To(model.Zones),
+				Tags:  pointer.To(model.Tags),
+			}
+
+			if err := client.CreateOrUpdateThenPoll(ctx, id, *properties); err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (r AMLFileSystemResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.AMLFileSystem.AmlFilesystems
+
+			id, err := amlfilesystems.ParseAmlFilesystemID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			var model AMLFileSystemModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			properties := amlfilesystems.AmlFilesystemUpdate{
+				Properties: &amlfilesystems.AmlFilesystemUpdateProperties{},
+			}
+
+			if metadata.ResourceData.HasChange("maintenance_window") {
+				properties.Properties.MaintenanceWindow = expandAMLFileSystemMaintenanceWindowForUpdate(model.MaintenanceWindow)
+			}
+
+			if metadata.ResourceData.HasChange("key_encryption_key") {
+				properties.Properties.EncryptionSettings = expandAMLFileSystemKeyEncryptionKey(model.KeyEncryptionKey)
+			}
+
+			if metadata.ResourceData.HasChange("tags") {
+				properties.Tags = pointer.To(model.Tags)
+			}
+
+			if err := client.UpdateThenPoll(ctx, *id, properties); err != nil {
+				return fmt.Errorf("updating %s: %+v", *id, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func (r AMLFileSystemResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.AMLFileSystem.AmlFilesystems
+
+			id, err := amlfilesystems.ParseAmlFilesystemID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			model := resp.Model
+			if model == nil {
```

Comment:
Instead of error out when `model` is nil, can we instead do [this](https://github.com/hashicorp/terraform-provider-azurerm/blob/22c6d8c1494c7301bcd02382ba55dd4cb9644c64/contributing/topics/guide-new-resource.md#:~:text=if%20model%20%3A%3D%20resp.Model%3B%20model%20!%3D%20nil%20%7B)?

Reply:
Updated